src/daft-csv/src/read.rs

```diff
@@ -54,57 +76,177 @@ async fn read_csv_single(
     include_columns: Option<Vec<&str>>,
     num_rows: Option<usize>,
     has_header: bool,
-    delimiter: Option<u8>,
+    delimiter: u8,
     io_client: Arc<IOClient>,
     io_stats: Option<IOStatsRef>,
+    schema: Option<SchemaRef>,
+    buffer_size: Option<usize>,
+    chunk_size: Option<usize>,
+    max_chunks_in_flight: Option<usize>,
```

Comment:
do we still need `max_chunks_in_flight` threaded all the way to the top now that we default to num cores?